BadApi/Controllers/Base/BaseApiController.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Data.SqlClient;
+using System.Linq;
+using System.Web;
+using System.Web.Http;
+
+namespace BadApi.Controllers.Base
+{
+	public class BaseApiController : ApiController
+	{
+		protected IDbConnection GetConnection()
+		{
+			var connection = new SqlConnection("Server=localhost;Database=BadApi;User Id=badapi_user;Password=p@ssw0rd;");
```

Comment:
Consider using dependency injection rather than direct instantiation of dependency in code. This method is not easily testable due to side effects and will contribute towards overall code being harder to maintain